### PR TITLE
feat: add oracle query expiration internal refund transfers

### PR DESCRIPTION
**Pull request by sborrazas**

### CODE REVIEW

File: priv/migrations/20221104120000_index_oracle_query_and_refunds.ex [hunk 1]
```diff
@@ -0,0 +1,155 @@
+defmodule AeMdw.Migrations.IndexOracleQueryAndRefunds do
```
Comment (jyeshe) [inline priv/migrations/20221104120000_index_oracle_query_and_refunds.ex:1]: Got this error with `testnet-1.25.1.tar.gz`:
```
warning: redefining module AeMdw.Migrations.IndexOracleQueryAndRefunds (current version loaded from _build/dev/lib/ae_mdw/ebin/Elixir.AeMdw.Migrations.IndexOracleQueryAndRefunds.beam)
  priv/migrations/20221104120000_index_oracle_query_and_refunds.ex:1

09:01:33.023 [error] CRASH REPORT Process <0.745.0> with 0 neighbours exited with reason: bad return value: {{'Elixir.AeMdw.Application',start_phase,[migrate_db,normal,[]]},{'EXIT',{function_clause,[{aeo_query_tx,oracle_id,[{ga_meta_tx,{id,account,<<96,70,239,88,30,239,116,157,73,35,96,177,84,44,123,233,151,181,221,202,13,46,81,10,67,18,178,23,153,139,252,116>>},<<43,17,244,119,202,45,27,255>>,3,50000,1000000000,80240000000000,0,{signed_tx,{aetx,oracle_query_tx,aeo_query_tx,188,{oracle_query_tx,{id,account,<<96,70,239,88,30,239,116,157,73,35,96,177,84,44,123,233,151,181,221,202,13,46,81,10,67,18,...>>},...}},...}}],...},...]}}} in application_master:init/4 line 138
```

**Review by jyeshe — CHANGES_REQUESTED** [review]

**Review by thepiwo — APPROVED** [review]

File: lib/ae_mdw/db/mutations/oracles_expiration_mutation.ex [hunk 2]
```diff
@@ -29,15 +32,32 @@ defmodule AeMdw.Db.OraclesExpirationMutation do
...
+    state2 =
+      state
+      |> Collection.stream(Model.ActiveOracleExpiration, {height, <<>>})
+      |> Stream.take_while(&match?({^height, _pk}, &1))
+      |> Enum.reduce(state, fn {^height, pubkey}, state ->
+        expire_oracle(state, height, pubkey)
+      end)
+
+    state2
+    |> Collection.stream(Model.OracleQueryExpiration, {height, <<>>})
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply lib/ae_mdw/db/mutations/oracles_expiration_mutation.ex]: Done

File: lib/ae_mdw/db/sync/oracle.ex [hunk 3]
```diff
@@ -37,28 +39,25 @@ defmodule AeMdw.Db.Sync.Oracle do
...
   end
 
-  @spec response_mutation(Node.tx(), Blocks.block_index(), Blocks.block_hash(), Txs.txi()) ::
-          OracleResponseMutation.t() | nil
-  def response_mutation(tx, block_index, block_hash, txi) do
+  @spec query_mutation(Node.tx(), height()) :: OracleQueryMutation.t()
+  def query_mutation(tx, height) do
+    oracle_pk = Validate.id!(:aeo_query_tx.oracle_id(tx))
+    ttl = :aeo_query_tx.ttl(tx)
+    expiration_height = height + ttl
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply lib/ae_mdw/db/sync/oracle.ex]: Done, now using the `query_id` with both `{:delta, tll} | `{:block, height}`

File: lib/ae_mdw/db/mutations/oracle_response_mutation.ex [hunk 4]
```diff
@@ -35,10 +40,16 @@ defmodule AeMdw.Db.OracleResponseMutation do
...
           txi: txi,
           oracle_pk: oracle_pk,
-          fee: fee
+          query_id: query_id
         },
         state
       ) do
-    IntTransfer.write(state, {height, txi}, "reward_oracle", oracle_pk, txi, fee)
+    Model.oracle_query(expire: expiration_height, fee: fee) =
+      State.fetch!(state, Model.OracleQuery, {oracle_pk, query_id})
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply lib/ae_mdw/db/mutations/oracle_response_mutation.ex]: Done, now handling these cases where the id doesn't exist. In reality, once we have the nonce fix from the node, all query IDs are going to be valid and all responses are going to have an equivalent query to fetch

File: priv/migrations/20221104120000_index_oracle_query_and_refunds.ex (hunk 1 above)
Comment (sborrazas) [reply priv/migrations/20221104120000_index_oracle_query_and_refunds.ex:1]: Done, handling for ga_meta added. Thanks!

**Review by jyeshe — APPROVED** [review]

**Pull request merged**